packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1305,12 +1305,47 @@ class PipelineOwner with DiagnosticableTreeMixin {
       return true;
     }());
     try {
-      final List<RenderObject> nodesToProcess = _nodesNeedingSemantics.toList()
+      // This has to be top-to-down order since the geometries of a child and its
+      // subtree depends on ancestors' transforms and clips. If it updates child
+      // first, it may use dirty geometry in parent's semantics node to
+      // calculate the geometries in the subtree.
+      final List<RenderObject> nodesToProcess = _nodesNeedingSemantics
+        .where((RenderObject object) => !object._needsLayout && object.owner == this)
```

Comment:
IIrc widget that are offstage won't layout during the layout stage, those renderobject may still end up here if they markNeedsSemanticsUpdate. We skip those since they won't participate in semantics tree anyway.